app/models/reservation_status.rb

```diff
@@ -1,2 +1,8 @@
 class ReservationStatus < ApplicationRecord
-end
\ No newline at end of file
+  # Create methods for retrieving all statuses directly
+  ['Booked', 'Confirmed', 'Cancelled'].each do |name|
+    singleton_class.send(:define_method, name.downcase) do
+      find_by(name: name)
+    end
+  end
```

Comment:
Not bad.

```ruby
%w[Booked Confirmed Cancelled].each do |name|
  define_method(name.downcase) do
    find_by(name: name)
  end
end
```

This won't work? Why do you need to call `singleton_class`?

Reply:
I used `singleton_class` because I am defining the static methods. Just `define_method` didn't work properly

Reply:
Ah right. So for such purpose you also have [define_singleton_method](https://ruby-doc.org/core-3.0.1/Object.html#method-i-define_singleton_method) which is almost the same but a little shorter.

Reply:
It works! Thanks! 🎉